components/tools/OmeroWeb/omeroweb/webgateway/views.py

```diff
@@ -774,8 +774,16 @@ def _get_prepared_image(request, iid, server_id=None, conn=None,
         return
     if 'c' in r:
         logger.debug("c="+r['c'])
-        channels, windows, colors, reverses = _split_channel_info(r['c'])
-        if not img.setActiveChannels(channels, windows, colors, reverses):
+        activechannels, windows, colors, reverses = _split_channel_info(r['c'])
+        allchannels = []
+        for ch in activechannels:
+            allchannels.append(abs(int(ch)))
+        # First save properties of all channels
+        if not img.setActiveChannels(allchannels, windows, colors, reverses):
```

Comment:
I added it with the last commit. But what happens now, is that the preview doesn't get updated any more. I don't think that's the expected behaviour.